docs/blog/2019-07-01-extending-components/index.md

```diff
@@ -0,0 +1,135 @@
+import Card from "../components/card"
+
+export default ({ name, bio, avatar, twitterUrl, githubUrl }) => (
+  <Card>
+    <MediaObject>
+      <Avatar {...avatar} />
+      <div>
+        <h3>{name}</h3>
+        <p>{bio}</p>
+        <a href={twitterUrl}>
+          <Icon name="twitter" />
+        </a>
+        <a href={githubUrl}>
+          <Icon name="github" />
+        </a>
+      </div>
+    </MediaObject>
+  </Card>
+)
+```
+
+This workflow works, especially since the component is relatively
+straightforward. However, it could be optimized in scenarios where
+you want to wrap a component or pass a different prop.
+
+### Importing the Shadowed Component
+
+In the above example it'd be preferable to be able to import the
+Author component and wrap it with your Card.
+
+Now, that component extending has been added you can do the
+following instead:
+
+```js:title=src/gatsby-theme-blog/components/author.js
+import React from "react"
+import { Author } from "gatsby-theme-blog/src/components/author"
+import Card from "../components/card"
+
+export default props => (
+  <Card>
+    <Author {...props} />
+  </Card>
+)
+```
+
+This is a quick and efficient way to quickly customize rendering
+without needing to worry about the implementation details of
+the component you're looking to customize. Extending the shadowed
+component means you can use composition, leveraging a great feature
+from React.
+
+### Applying New Props
+
+In some cases there could be a component with different variants
+without an API to modify it outside of shadowing. With component
+extending you can import that component and then add your new
+prop to change it.
+
+```js:title=src/gatsby-theme-blog/components/newsletter/button.js
+import { NewsletterButton } from "gatsby-theme-blog/src/components/newsletter"
+
+export default props => <NewsletterButton {...props} variant="link" />
+```
+
+### Using the CSS Prop
+
+In addition to passing a different prop to a component you're extending,
+you might want to apply CSS using the [Emotion CSS prop](/docs/emotion).
+This will allow you to change the styling of a particular component without
+changing any of its functionality.
+
+```js:title=src/gatsby-theme-blog/components/newsletter/button.js
+import { NewsletterButton } from "gatsby-theme-blog/src/components/newsletter"
+
+export default props => (
+  <NewsletterButton
+    css={{
+      backgroundColor: "rebeccapurple",
+      color: "white",
+      boxShadow: "none",
+    }}
+    {...props}
+  />
+)
+```
+
+### Change an Object
+
+Another use case might be changing a `theme.js` file that a Gatsby
+Theme uses to set colors globally.
+
+```js:title=src/gatsby-theme-blog/theme.js
+import theme from "gatsby-theme-blog/src/theme"
+
+export default { ...theme, primary: "tomato" }
+```
+
+This provides a nice interface to extend an object if you want to
+change a couple values from the defaults.
+
+## Conclusion
+
+If you need to make a small change to an existing component or
```

Comment:
When are we hoping to publish this post? Can we allude to what's coming with regards to stable themes + docs?

Reply:
I _think_ we could probably publish this today. Would definitely be nice to be able to link to it in upcoming posts.

> Can we allude to what's coming with regards to stable themes + docs?

Totally! Though not exactly sure how we should do that. 🤔

Reply:
I think go ahead and publish as-is, and we could go back and add a link to the Themes docs later since they aren't quite ready yet.